Modules/_functoolsmodule.c

```diff
@@ -70,23 +147,30 @@ get_functools_state_by_type(PyTypeObject *type)
 static PyObject *
 partial_new(PyTypeObject *type, PyObject *args, PyObject *kw)
 {
-    PyObject *func, *pargs, *nargs, *pkw;
+    PyObject *func, *pto_args, *new_args, *pto_kw;
     partialobject *pto;
+    Py_ssize_t pto_phcount = 0;
+    Py_ssize_t new_nargs = PyTuple_GET_SIZE(args);
 
-    if (PyTuple_GET_SIZE(args) < 1) {
+    if (new_nargs < 1) {
         PyErr_SetString(PyExc_TypeError,
                         "type 'partial' takes at least one argument");
         return NULL;
     }
+    new_nargs--;
```

Comment:
I don't really like removing one here just to avoid some -1 afterwards.

Reply:
I tried to do it as early as possible, because `new_nargs` means new number of positional arguments.

I went the other way...:
```
Py_ssize_t new_nargs = PyTuple_GET_SIZE(args) - 1;
...
if (new_nargs < 0) {
```

Reply:
> because new_nargs means new number of positional arguments.

Ok now I understand more. Actually, it's not changing the value that bothered me but rather changing the value *here* (and using `--`!). The next usage for this variable is 100 after that which is quite far away...